components/Popover.php

```diff
@@ -0,0 +1,506 @@
+	 * @var bool
+	 */
+	protected $popover_close_outside = true;
+
+	/**
+	 * Set Popover title
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $popover_title the popover title.
+	 *
+	 * @return self
+	 */
+	public function title( string $popover_title ): self {
+		$this->popover_title = $popover_title;
+		return $this;
+	}
+
+	/**
+	 * Set Popover body html.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $popover_body the popover body html.
+	 *
+	 * @return self
+	 */
+	public function body( string $popover_body ): self {
+		$this->popover_body = $popover_body;
+		return $this;
+	}
+
+	/**
+	 * Set popover placement position.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $popover_placement the placement position.
+	 *
+	 * @return self
+	 */
+	public function placement( string $popover_placement = 'bottom-start' ): self {
+		$placement_positions = array( 'top', 'left', 'right', 'bottom', 'bottom-start' );
+		if ( ! in_array( $popover_placement, $placement_positions, true ) ) {
+			$this->popover_placement = 'bottom-start';
+		}
+
+		$this->popover_placement = $popover_placement;
+		return $this;
+	}
+
+	/**
+	 * Popover button component.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $popover_button the popover button element html.
+	 *
+	 * @return self
+	 */
+	public function trigger( string $popover_button ): self {
+		$this->popover_button = $popover_button;
+		return $this;
+	}
+
+	/**
+	 * Whether to hide close button on popover.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param boolean $show_close_button determines whether to hide close button.
+	 *
+	 * @return self
+	 */
+	public function closeable( bool $show_close_button = false ): self {
+		$this->show_close_button = $show_close_button;
+		return $this;
+	}
+
+	/**
+	 * Set the popover footer alignment.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $popover_footer_alignment the popover footer alignment class.
+	 *
+	 * @return self
+	 */
+	public function footer_alignment( string $popover_footer_alignment ): self {
+		$this->popover_footer_alignment = $popover_footer_alignment;
+		return $this;
+	}
+
+	/**
+	 * Set popover footer buttons.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param array $popover_footer_buttons the footer button list.
+	 *
+	 * @return self
+	 */
+	public function footer( array $popover_footer_buttons = array() ): self {
+		$this->popover_footer_buttons = $popover_footer_buttons;
+		return $this;
+	}
+
+	/**
+	 * Menu items for creating popover menu.
+	 *
+	 * @since 4.0.0
+	 *
+	 * @param string $tag the menu item html tag.
+	 * @param string $popover_menu_item the popover menu item.
+	 * @param string $class the menu item class.
+	 * @param string $icon the menu item icon.
+	 * @param string $icon_alignment menu item icon alignment.
+	 * @param array  $attr the menu item attributes.
+	 *
+	 * @return self
+	 */
+	public function menu_item( string $tag, string $popover_menu_item, string $class = '', string $icon = '', string $icon_alignment = '', array $attr = array() ): self {
```

Comment:
Instead of passing a long list of args, make it an array argument and add a comment about the array definition.